### PR TITLE
Add static memory config

### PR DESCRIPTION
**Pull request by chinglee-iot**
Remove the usage of malloc/free in cellular interface.
* Use stack variable instead of malloc in Cellular_CommonGetRegisteredNetwork
* Use stack variable instead of Cellular_ATStrDup in _processUrcPacket
* Add config, CELLULAR_CONFIG_STATIC_AT_RESPONSE, to allocate static allocated cellularStaticResponse and cellularStaticATCommandLineTable instead of malloc.
* Add new test cases to cover allocate AT response fail and AT command line fail.

User of FreeRTOS Cellular Interface can enable the following configs to prevent malloc/free in cellular common
* CELLULAR_CONFIG_STATIC_ALLOCATION_CONTEXT ( default set to 1 )
* CELLULAR_CONFIG_STATIC_SOCKET_CONTEXT_ALLOCATION ( default set to 1 )
* CELLULAR_CONFIG_STATIC_AT_RESPONSE ( default set to 1 )

### CODE REVIEW

File: source/cellular_common.c [hunk 1]
```diff
@@ -542,47 +540,35 @@ CellularError_t _Cellular_CreateSocketData( CellularContext_t * pContext,
...
 
     taskEXIT_CRITICAL();
 
-    if( cellularStatus == CELLULAR_NO_MEMORY )
-    {
-        LogError( ( "_Cellular_CreateSocket, Out of memory" ) );
-    }
-    else if( socketId >= CELLULAR_NUM_SOCKET_MAX )
-    {
-        LogError( ( "_Cellular_CreateSocket, No free socket slots are available" ) );
```
Comment (ActoryOu) [inline source/cellular_common.c:578]: I suggest to keep this error message if no free socket ID found for debugging.

File: source/cellular_pktio.c [hunk 2]
```diff
@@ -273,24 +302,65 @@ static CellularATCommandResponse_t * _Cellular_AtResponseNew( void )
...
+{
+    CellularATCommandLine_t * pNew = NULL;
 
-            /* Ruese the pktiobuffer. No need to free pToFree->pLine here. */
-            Platform_Free( pToFree );
+    #if ( CELLULAR_CONFIG_STATIC_AT_RESPONSE == 1 )
+        if( cellularStaticATCommandIndex < CELLULAR_NUM_AT_COMMAND_LINE )
+        {
+            pNew = &cellularStaticATCommandLineTable[ cellularStaticATCommandIndex ];
+            cellularStaticATCommandIndex++;
```
Comment (ActoryOu) [inline source/cellular_pktio.c:344]: Is it possible that multi-threads request to send AT command at the same time?
Do we need to handle this race condition?

File: source/cellular_common.c (hunk 1 above)
Comment (chinglee-iot) [reply source/cellular_common.c:578]: Update in new commit.

**Review by ActoryOu — APPROVED** [review]

**Comment by chinglee-iot** [comment]
This PR need to be updated. Close this PR first.

**Pull request closed without merging**